desktop/apps/auction2/components/artwork_browser/main/ArtworkDisplay.js

```diff
@@ -2,106 +2,128 @@ import Jump from 'desktop/components/jump/react'
 import MasonryArtwork from 'desktop/apps/auction2/components/artwork_browser/main/artwork/MasonryArtwork'
 import GridArtwork from 'desktop/apps/auction2/components/artwork_browser/main/artwork/GridArtwork'
 import ListArtwork from 'desktop/apps/auction2/components/artwork_browser/main/artwork/ListArtwork'
-import LoadingSpinner from 'desktop/apps/auction2/components/artwork_browser/main/artwork/LoadingSpinner'
+import LoadingSpinner from 'desktop/apps/auction2/components/artwork_browser/main/LoadingSpinner'
 import MasonryGrid from 'desktop/components/react/masonry_grid/MasonryGrid'
 import PropTypes from 'prop-types'
-import React from 'react'
+import React, { Component } from 'react'
 import get from 'lodash.get'
 import { connect } from 'react-redux'
 
-function ArtworkDisplay ({
-  isFetchingArtworks,
-  isListView,
-  isMobile,
-  isClosed,
-  saleArtworks,
-  saleId
-}) {
-  const listType = isListView ? '--list' : ''
-
-  return (
-    <div className={`auction2-page-artworks${listType}`}>
-      {(() => {
-        if (isMobile) {
-          if (isListView) {
-            return (
-              <div>
-                { saleArtworks.map((saleArtwork, key) => (
-                  <ListArtwork
-                    artwork={saleArtwork}
-                    key={key}
+class ArtworkDisplay extends Component {
+  static propTypes = {
+    isFetchingArtworks: PropTypes.bool.isRequired,
+    isListView: PropTypes.bool.isRequired,
+    isMobile: PropTypes.bool.isRequired,
+    saleArtworks: PropTypes.array.isRequired
+  }
+
+  state = {
+    isMounted: false,
+    showFullScreenSpinner: false
+  }
+
+  componentDidMount () {
+    this.setState({
+      isMounted: true
```

Comment:
**Note**: The only way to implement SSR renders with client-side conditional checks (`typeof window !== 'undefined'`) is to set state, otherwise you get a React error stating that the markup checksum differs between client and server. This is necessary since we're only adding the overlay preloader if the user has scrolled within a particular region (`window.scrollY`).